File: java/monkey/src/main/java/root/ast/Identifier.java
package root.ast;

import root.Token;

public class Identifier extends Expression {
    public Token token;
    public String value;

    public Identifier(Token token, String value) {
        this.token = token;
        this.value = value;
    }

    @Override
    public void expressionNode() {
    }

    @Override
    public String tokenLiteral() {
        return token.literal();
    }
}

File: php/src/ast/expressions/Identifier.php
<?php

require 'vendor/autoload.php';

readonly class Identifier implements Expression
{
    public function __construct(
        public Token $token,
        public string $value
    ) {
    }

    public function tokenLiteral(): string
    {
        return $this->token->literal;
    }

    public function expressionNode(): void
    {
    }

    public function __toString(): string
    {
        return $this->value;
    }
}